eclair-core/src/main/scala/fr/acinq/eclair/payment/send/Recipient.scala

```diff
@@ -164,9 +164,13 @@ case class BlindedRecipient(nodeId: PublicKey,
   }
 
   override def buildPayloads(paymentHash: ByteVector32, route: Route): Either[OutgoingPaymentError, PaymentPayloads] = {
-    validateRoute(route).map(blindedHop => {
-      val blindedPayloads = buildBlindedPayloads(route.amount, blindedHop)
-      Recipient.buildPayloads(blindedPayloads, route.hops)
+    validateRoute(route).flatMap(blindedHop => {
+      if (route.hops.length > 1 && blindedHop.resolved.route.isInstanceOf[BlindedPathsResolver.PartialBlindedRoute]) {
+        Left(IndirectRelayInBlindedRoute(blindedHop.nodeId))
+      } else {
```

Comment:
nit: this should be done in `validateRoute`, this is part of route validation?

Reply:
Done